platform/engineering/design-docs/2021-01-13-web-component-library.md

```diff
@@ -0,0 +1,221 @@
+
+#### Migration to Web Components
+
+Once a new release is made for the Web Component library, we will [configure a linting rule](https://github.com/AlexMost/eslint-plugin-deprecate/blob/master/docs/rules/import.md) as part of the `lint:js:changed` script in `vets-website` to throw an error if a React component is imported that has a released Web Component replacement. This will prevent the older React component from appearing in more places as we are actively making efforts to remove it.
+
+Next, if we believe a particular component migration will be problematic the Design System team will reach out to a VFS team to work with them on the migration. Once the first team has completed their migration to the new Component, we will announce the success and encourage other teams to do the same. At this point we will provide support for the migration if requested, but we will no longer provide support for the legacy React component.
+
+After the new Web Component has replaced all instances of its React/`formation`/liquid conterparts, the React component will be removed from `component-library` and a new version will be published. The associated styles will also be removed from `formation` if we are certain that this will not affect other styles on va.gov.
+
+Additionally, the Storybook stories for the React component will be replaced by a simple document that points the user to a different Storybook host that is exclusively for Web Components. See [Chromatic hosting](#chromatic-hosting) for possible future work to improve on this.
+
+This process will repeat for each Component that has React/`formation`/liquid counterparts to replace.
+
+### Code Location
+
+The code will live in the `component-library` repo on a dedicated orphan branch named `web-components`. This code will be used in `vets-website` via a dependency in `package.json`
+
+### Testing Plan
+
+Each Component will have:
+
+- Unit and E2E tests written using Jest ([provided by Stencil](https://stenciljs.com/docs/testing-overview))
+- One or more stories in Storybook to validate appearance/behavior
+
+As of mid-January 2021, Stencil provides an experimental screenshot visual diff testing functionality. If this becomes stable, we will consider adopting it.
+
+### Logging
+
+N/A
+
+### Debugging
+
+Storybook will be a useful tool for debugging and development. Stencil also comes with its own development server which can be used.
+
+There will be no special provisions for debugging provided for consumers of the WC library. Right now there is no plan to implement Sentry logging in the components.
+
+### Caveats
+
+To be determined.
+
+### Security Concerns
+
+N/A
+
+### Privacy Concerns
+
+N/A
+
+### Open Questions and Risks
+
+How will we ensure that each Component meets accessibility standards?
+
+How will we measure performance of Web Components, lazy-loading, etc?
+
+#### Risks
```

Comment:
Another good point @drorva mentioned to me is industry adoption. My understanding is it's been pretty slow going heretofore, so if the standard ends up becoming obsolete, that may affect the future of this work. I don't know what the risk of that is, but it's worth calling out.

Reply:
Included that point in this commit: https://github.com/department-of-veterans-affairs/va.gov-team/pull/18662/commits/86772e69b3b2d04821684374714f1df6c16135d6